src/rpc/xmlrpc.cc

```diff
@@ -607,6 +613,357 @@ XmlRpc::set_size_limit(uint64_t size) {
   xmlrpc_limit_set(XMLRPC_XML_SIZE_LIMIT_ID, size);
 }
 
+bool
+XmlRpc::is_valid() const { return m_env != NULL; }
+
+#elif defined(HAVE_XMLRPC_TINYXML2)
+
+class xmlrpc_error : public torrent::base_error {
+public:
+  xmlrpc_error(int type, std::string msg) : m_type(type), m_msg(msg) {}
+  virtual ~xmlrpc_error() throw() {}
+
+  virtual int         type() const throw() { return m_type; }
+  virtual const char* what() const throw() { return m_msg.c_str(); }
+
+private:
+  int                 m_type;
+  std::string         m_msg;
+};
+
+const tinyxml2::XMLElement* element_access(const tinyxml2::XMLElement* elem, std::string elemNames) {
+  // Helper function to check each step of a element access, in liueu of XPath
+  char token = ',';
+  std::stringstream ss(elemNames);
+  std::string item;
+  std::string stack("");
+  const tinyxml2::XMLElement* result = elem;
+  while (getline (ss, item, token)) {
```

Comment:
Using stringstream is inefficient, especially when you know you're searching for a single-char token.

Iterators and std::find should be enough.